pkg/mock/dynamic/dynamic.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package dynamic
+
+import (
+	"log"
+	"net/http"
+	"reflect"
+
+	cfg "github.com/aws/amazon-ec2-metadata-mock/pkg/config"
+	"github.com/aws/amazon-ec2-metadata-mock/pkg/mock/imdsv2"
+	"github.com/aws/amazon-ec2-metadata-mock/pkg/server"
+)
+
+var (
+	supportedPaths   = make(map[string]interface{})
+	response         interface{}
+	jsonTextResponse = map[string]bool{"": true}
```

Comment:
according to [aws docs](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-categories.html) `instance-identity/document` returns a "**JSON** containing instance attributes,..."

`jsonTextResponse` and accompanying logic can be removed if the response is truly a JSON

Reply:
according to the template Handler() code, I only remove the map. And leave the jsonTextResponse variable for future usage.